### DISCUSSION TITLE
How should I get application folder correctly in Linux DRM?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#15625] [Q&A] [answered]
I want to get the folder of my Avalonia application.

My Avalonia application is located in /home/admin/AvaloniaApplication

At first, I obtained it through AppDomain.CurrentDomain.BaseDirectory. It can obtain the path normally on Windows, but the path obtained on Linux is always /home/admin/

Then I tried to use AppContext.BaseDirectory, but I also got the incorrect path /home/admin/

I saw on the Internet that Path.GetDirectoryName(Assembly.GetEntryAssembly().Location) is used to obtain it, but this method obtains a blank path.

Now I can't think of any better way to solve it. Could you please give me a good suggestion? Thank you.

**Comment by thevortexcloud** [comment, marked as answer]
Every single one of these returns the same correct path for me:

```csharp
        Console.WriteLine(Directory.GetCurrentDirectory());
        Console.WriteLine(AppContext.BaseDirectory);
        Console.WriteLine(AppDomain.CurrentDomain.BaseDirectory);
```

Technically the current directory can change after the app starts, but almost nobody does that in .NET.

> I saw on the Internet that Path.GetDirectoryName(Assembly.GetEntryAssembly().Location) is used to obtain it, but this method obtains a blank path.

According to Scott Hanselman (a guy who works for Microsoft):

> How does it work in .NET Core 5 (> preview 7)? Well, because the assemblies are embedded and loaded from memory in 5, so like any other in-memory assembly, they don't have a location. Any use of Assembly.Location is going to be suspect. 

https://www.hanselman.com/blog/how-do-i-find-which-directory-my-net-core-console-application-was-started-in-or-is-running-from

Are you sure you are not accidentally launching from the wrong location? Is `AvaloniaApplication` the name of your actual app or is it meant to be a folder?

**Reply by kyuranger** [reply]
The problem is solved, it was my code that had the problem and not the actual situation mentioned in my question. But through this time, I also learned that it would be more appropriate to use AppContext.BaseDirectory. Thank you very much.
